sdk/test/trace/simple_processor_test.cc

```diff
@@ -0,0 +1,67 @@
+#include "src/trace/simple_processor.h"
+#include "opentelemetry/nostd/span.h"
+#include "opentelemetry/sdk/trace/span_data.h"
+
+#include <gtest/gtest.h>
+
+using namespace opentelemetry::sdk::trace;
+
+/**
+ * A mock exporter that switches a flag once a valid recordable was received.
+ */
+class MockSpanExporter final : public SpanExporter
+{
+public:
+  MockSpanExporter(std::shared_ptr<bool> span_received,
+                   std::shared_ptr<bool> shutdown_called) noexcept
+      : span_received_(span_received), shutdown_called_(shutdown_called)
+  {}
+
+  std::unique_ptr<Recordable> MakeRecordable() noexcept override
+  {
+    return std::unique_ptr<Recordable>(new SpanData);
+  }
+
+  ExportResult Export(
+      opentelemetry::nostd::span<std::unique_ptr<Recordable>> &spans) noexcept override
+  {
+    for (auto span = spans.begin(); span != spans.end(); ++span)
+    {
+      auto raw = span->release();
+      if (static_cast<SpanData *>(raw))
+      {
+        *span_received_ = true;
+      }
+      delete raw;
+    }
```

Comment:
This would accomplish the same thing
```
for (auto& span : spans) {
  if (span != nullptr) {
     *span_received_ = true;
  }
}
```
static_cast isn't a checked cast so `if(static_cast<SpanData*>(span.get()) != nullptr)` is the same as `if(span != nullptr)`.

Reply:
That slipped in, it's fixed now.